### PR TITLE
Search API endpoint

### PR DESCRIPTION
**Pull request by colinyoung87**

### CODE REVIEW

File: app/controllers/api/v1/search_controller.rb [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+
+        results = SearchDocument.search(term, current_team.id, current_user.project_ids).page(page).per(limit)
+
+        has_more_results = (page * limit) < results.total_count
+        next_url = api_v1_search_index_path(term: term, page: page + 1, limit: limit) if has_more_results
+
+        meta = {
+          term: term,
+          total: results.total_count,
+          next_url: next_url
```
Comment (phawk) [inline app/controllers/api/v1/search_controller.rb]: would be nice to put the `has_more` boolean into the json

File: app/controllers/api/v1/search_controller.rb [hunk 2]
```diff
@@ -1,6 +1,40 @@
...
+        }
+
+        render json: results, meta: meta, each_serializer: SearchDocumentApiSerializer
```
Comment (phawk) [inline app/controllers/api/v1/search_controller.rb]: First time I’ve seen that meta thing for AMS :+1:

File: app/serializers/search_document_api_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+class SearchDocumentApiSerializer < ActiveModel::Serializer
+  attributes :searchable_id, :searchable_type, :searchable, :url, :project, :project_url, :story, :story_url, :attachment_image
+
+  def project_url
+    project.present? ? api_v1_project_path(project) : nil
```
Comment (phawk) [inline app/serializers/search_document_api_serializer.rb]: Does our search document preload project and story so this query isn’t going to N+1 ?

**Comment by phawk** [comment]
:+1:

File: app/serializers/search_document_api_serializer.rb (hunk 3 above)
Comment (colinyoung87) [inline app/serializers/search_document_api_serializer.rb]: aye, it does :)

**Pull request merged**